### ISSUE TITLE
Unable restore server from successful rear backup 

### ISSUE THREAD

**Issue by sameergithub5** [rear/rear#1445]
#### Relax-and-Recover (ReaR) Issue Template

Fill in the following items before submitting a new issue
(quick response is not guaranteed with free support):

* rear version (/usr/sbin/rear -V): Realx-and-Recover 2.00/ Git
* OS version (cat /etc/rear/os.conf or lsb_release -a):LSB package is not installed
* rear configuration files (cat /etc/rear/site.conf or cat /etc/rear/local.conf):
<pre>
OUTPUT=ISO
OUTPUT_URL=nfs://192.168.211.128/shared
BACKUP=NETFS
BACKUP_URL=nfs://192.168.211.128/shared
</pre>
* Are you using legacy BIOS or UEFI boot?
legacy BIOS
* Brief description of the issue:
Unable to recover 7.2 centos says mount failed

* Work-around, if any:
NO
<img width="515" alt="vm error" src="https://user-images.githubusercontent.com/21688051/29325074-0d16898e-8204-11e7-8df5-e61179917709.PNG">

**Comment by gozora** [comment]
Hi @sameergithub5,

Maybe you could share _rear-qa.log_ with us?
If you try to mount fileystem manually, like:
```
mkdir /tmp/test && \
mount -v -t nfs 192.168.211.128:/shared /tmp/test
```
what does the output say?

V.

**Comment by sameergithub5** [comment]
This is the output with nfsver4.

[root@qa ~]# mkdir /tmp/test && \
> mount -v -t nfs 192.168.211.128:/shared /tmp/test
mount.nfs: timeout set for Wed Aug 16 18:36:06 2017
mount.nfs: trying text-based options 'vers=4,addr=192.168.211.128,clientaddr=192.168.211.130'
[root@qa ~]# cd /tmp/test/
[root@qa test]# ll
total 760
drwxr-x--- 2 nfsnobody nfsnobody    106 Aug 16 03:29 qa
drwxr-xr-x 7 root      root        4096 Jan 20  2017 rear-master
-rw-r--r-- 1 root      root      772599 Mar  4 14:03 rear-master.zip
[root@qa test]# df -kh
Filesystem               Size  Used Avail Use% Mounted on
/dev/mapper/centos-root   13G  4.5G  8.5G  35% /
devtmpfs                 903M     0  903M   0% /dev
tmpfs                    913M     0  913M   0% /dev/shm
tmpfs                    913M  8.6M  904M   1% /run
tmpfs                    913M     0  913M   0% /sys/fs/cgroup
/dev/sda1                497M  125M  373M  26% /boot
tmpfs                    183M     0  183M   0% /run/user/0
192.168.211.128:/shared   28G  5.3G   23G  20% /tmp/test
[root@qa test]#

**Comment by gozora** [comment]
Hmm, no reason for ReaR return error on mount as it is visible from you screenshot.
Can you please run `rear -d -D recover` and provide us logfile from failed session? (/var/log/rear/rear-qa.log)

**Comment by sameergithub5** [comment]
[root@qa sbin]# ./rear -v -d -D mkbackup
Relax-and-Recover 2.00 / Git
Using log file: /tmp/rear-master/var/log/rear/rear-qa.log
Using backup archive '/tmp/rear.Bn5U8Z0wIODQI9L/outputfs/qa/backup.tar.gz'
Creating disk layout
Creating root filesystem layout
Copying logfile /tmp/rear-master/var/log/rear/rear-qa.log into initramfs as '/tmp/rear-qa-partial-2017-08-16T05:08:30+0530.log'
Copying files and directories
Copying binaries and libraries
Copying kernel modules
Creating initramfs
Making ISO image
Wrote ISO image: /tmp/rear-master/var/lib/rear/output/rear-qa.iso (109M)
Copying resulting files to nfs location
Saving /tmp/rear-master/var/log/rear/rear-qa.log as rear-qa.log to nfs location
Creating tar archive '/tmp/rear.Bn5U8Z0wIODQI9L/outputfs/qa/backup.tar.gz'
Archived 495 MiB [avg 5898 KiB/sec] OK
Archived 495 MiB in 87 seconds [avg 5830 KiB/sec]
You should also rm -Rf /tmp/rear.Bn5U8Z0wIODQI9L
[root@qa sbin]# 
=============================================
Kindly find attached 
/tmp/rear-master/var/log/rear/rear-qa.log file for backup happened.
so that u can spot any error in backup which is causing incorrect restoration.
************************************************************************
Restoration:
I have attached screenshot of rear -d -D recover.
<img width="529" alt="restore time error" src="https://user-images.githubusercontent.com/21688051/29364202-597fa11e-82b0-11e7-8f05-e96ee9b276dd.PNG">

I am not able to find file /var/log/rear/rear-qa.log

**Comment by gozora** [comment]
Your `rear mkbackup` looks to be working fine so I don't need logs from this session.

What you need to do to locate rear-qa.log from `rear recover` is following:
1. Boot your ReaR rescue/recover system from ISO file
2. Run `rear -d -D recover`
3. Your log file is created in _/var/log/rear/_  directory, so you need to download it and post it to Github ...

V.

**Comment by gozora** [comment]
@sameergithub5 for the case you don't know it yet, you can login to ReaR rescue/recovery system using SSH (c.f. [default.conf](https://github.com/rear/rear/blob/master/usr/share/rear/conf/default.conf#L767)).

**Comment by sameergithub5** [comment]
<img width="867" alt="last restore log" src="https://user-images.githubusercontent.com/21688051/29365886-0f89ae46-82b6-11e7-9bbb-b9cbfcb3a299.PNG">
Kindly see attached /var/log/rear/rear-qa.log

**Comment by schabrolles** [comment]
@sameergithub5,  just before starting a ReaR recovery, could you please run the following command:
- `ip a`
- `route -n`
May be your network configuration is not correctly set....

**Comment by sameergithub5** [comment]
Kindly find attached screenshot
<img width="518" alt="ip route details" src="https://user-images.githubusercontent.com/21688051/29372649-5dd9ba4c-82c9-11e7-89c0-a9201c3650e4.PNG">

**Comment by gozora** [comment]
@schabrolles was right, your network interface is not configured. You are probably trying to restore your OS to other then original server.
All you need to do is configure your NIC:
`ip address add <IP>/<NETMASK_PREFIX> dev eno16777736` and run `rear recover`

V.

**Comment by schabrolles** [comment]
I think #1349 should solved this one (ReaR 2.1)

**Comment by sameergithub5** [comment]
Where can i find the link for downloading this bug fix version.

**Comment by schabrolles** [comment]
@sameergithub5 
you can try to get it from here : https://github.com/rear/rear/releases/tag/2.2

**Comment by sameergithub5** [comment]
Hello Guyz,
I have downloaded 2.2 git rear version but story remains the same.
server still not reachable


Please see attached screenshot.
<img width="559" alt="2 2 rear error" src="https://user-images.githubusercontent.com/21688051/29412869-11965d16-8378-11e7-9d09-3238f4649435.PNG">

**Comment by sameergithub5** [comment]
I manually tried to configured but still no luck
<img width="524" alt="configure network" src="https://user-images.githubusercontent.com/21688051/29413152-044bc32a-8379-11e7-96ad-553bead18162.PNG">

**Comment by gozora** [comment]
Are you sure your network address mask is 255.255.255.255 ?

**Comment by gozora** [comment]
My best guess is that following could work much better `ip a add 192.168.211.130/24 dev eno16777736`

**Comment by gdha** [comment]
@sameergithub5 Do not forget your hypervisor virtual network - are you in the same VLAN?

**Comment by jsmeix** [comment]
To set up networking manually in the ReaR recovery system
one can use NETWORKING_PREPARATION_COMMANDS,
see usr/share/rear/conf/default.conf

**Comment by gdha** [comment]
@sameergithub5 Great that your problem has been fixed. So, this issue can be closed then?

**Comment by sameergithub5** [comment]
Yes absolutely . thanks 4 it
